problem_builder/answer.py

```diff
@@ -115,6 +115,20 @@ def add_error(msg):
         if not data.name:
             add_error(u"A Question ID is required.")
 
+    def student_view_user_state(self, context=None):
```

Comment:
We need `student_view_data` to appear in the course blocks API (as it already does), but `student_view_user_state` does not need to appear in the course blocks API. Instead we want `student_view_user_state` to be a handler, which we call just before displaying the block. It might make more sense to call it `get_student_view_user_state`, which is a more handler-like name.